app/src/main/java/com/example/caregiver/BeaconAddRegion.java

```diff
@@ -70,45 +73,80 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         View rootView = inflater.inflate(R.layout.fragment_beacon_add_region, container, false);
 
         displayUuid(rootView, userId);
-        Button updateButton = (Button) rootView.findViewById(R.id.uuidUpdateButton);
+        Button updateButton = rootView.findViewById(R.id.uuidUpdateButton);
         updateButton.setOnClickListener(v -> updateUuid(user, rootView));
 
-        Button addRegionButton = (Button) rootView.findViewById(R.id.add_region);
+        Button addRegionButton = rootView.findViewById(R.id.add_region);
         addRegionButton.setOnClickListener(v -> addRegion(user, rootView));
 
         return rootView;
     }
 
+    public String getUUIDValue(View rootView) {
+        UUIDField = rootView.findViewById(R.id.UUIDText);
+        return UUIDField.getText().toString();
+    }
+
+    public String getRegionName(View rootView) {
+        regionNameField = rootView.findViewById(R.id.regionName);
+        return regionNameField.getText().toString();
+    }
+
+    public String getMajorValue(View rootView) {
+        majorField = rootView.findViewById(R.id.major);
+        return majorField.getText().toString();
+    }
+
     @RequiresApi(api = Build.VERSION_CODES.O)
     public void addRegion(@NonNull FirebaseUser user, View rootView) {
+        BeaconRegionList.getInstance().getUsersRegions(new BeaconRegionList.OnQueryCompleteListener() {
+            @Override
+            public void onSuccess(DataSnapshot dataSnapshot) {
+                BeaconRegionList.getInstance().createRegionMajorMap(dataSnapshot);
+                Log.i("sample", "regionMajorMap after the getUsersRegion call" + BeaconRegionList.regionMajorMap.toString());
+                String UUIDValue = getUUIDValue(rootView);
 
-        UUIDField = (EditText) rootView.findViewById(R.id.UUIDText);
-        String UUIDValue = UUIDField.getText().toString();
+                String regionName = getRegionName(rootView);
 
-        regionNameField = (EditText) rootView.findViewById(R.id.regionName);
-        String regionName = regionNameField.getText().toString();
+                String majorValue = getMajorValue(rootView);
 
-        majorField = (EditText) rootView.findViewById(R.id.major);
-        String majorValue = majorField.getText().toString();
+                if (regionName.isEmpty() || majorValue.isEmpty()) {
+                    displayErrorMessage("One or more fields are empty.", rootView);
+                }
+                // check if region/major value has already been used
+                else if (BeaconRegionList.regionMajorMap.containsKey(regionName.toLowerCase()) || BeaconRegionList.regionMajorMap.containsValue(majorValue)) {
```

Comment:
I wonder if it will help to be more explicit with this error message. Instead of name or value is used, we can say "name is used" and "value is used". This can be useful for the user to correct the mistake quickly.